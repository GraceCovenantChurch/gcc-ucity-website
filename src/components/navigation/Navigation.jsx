import React from "react";

import { Link } from "react-router-dom";
import { useMediaQuery } from "react-responsive";

import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";

import HamburgerMenu from "./menu/HamburgerMenu";
import NavigationMenu from "./menu/NavigationMenu";

import { MOBILE_QUERY } from "constants/mobile";

import imageURLDesktop from "static/gcclogo.png";
import imageURLMobile from "static/gcclogo-black.png";

const useStyles = makeStyles((theme) => ({
  navRoot: {
    flexGrow: 1,
    // margin: theme.spacing(3),
    position: "absolute",
    width: "100vw",
    fontFamily: "IBM Plex Sans",
  },
  menuDesktopButton: {
    marginRight: theme.spacing(5),
    height: 85,
  },
  menuMobileButton: {
    marginRight: theme.spacing(2),
    marginTop: ".75vh",
    height: 70,
  },
  mobileToolBar: {
    margin: ".5em",
    padding: 0,
  },
  titleGrow: {
    flexGrow: 1,
  },
  appBarDesktop: {
    padding: "1.5em",
    background: "transparent",
    boxShadow: "none",
  },
  appBarMobile: {
    background: "white",
    // boxShadow: "none",
  },
}));

const Navigation = (props) => {
  const classes = useStyles();
  const isMobile = useMediaQuery({ query: MOBILE_QUERY });

  const menuOptions = [
    {
      title: "About",
      type: "dropdown",
      children: [
        { title: "Welcome", type: "link", children: "/welcome" },
        { title: "Beliefs", type: "link", children: "/beliefs" },
        { title: "Staff", type: "link", children: "/staff" },
        {
          title: "Main Line Site",
          type: "link",
          children: "https://ml.gracecovenant.net",
        },
        { title: "AMI", type: "link", children: "https://amichurches.com/" },
      ],
    },
    { title: "Family Groups", type: "link", children: "/familygroup" },
    // { title: "Ministries", type: "link", children: "/ministries" },
    { title: "Giving", type: "link", children: "/giving" },
    { title: "Multimedia", type: "link", children: "/multimedia" },
    { title: "Events", type: "link", children: "/events" },
  ];

  return (
    <React.Fragment>
      <AppBar
        className={isMobile ? classes.appBarMobile : classes.appBarDesktop}
        position={isMobile ? "sticky" : "absolute"}
      >
        <Toolbar className={isMobile ? classes.mobileToolBar : ""}>
          <div className={classes.titleGrow}>
            <Link to="/">
              <img
                alt="nav-logo"
                className={
                  isMobile
                    ? classes.menuMobileButton
                    : classes.menuDesktopButton
                }
                src={isMobile ? imageURLMobile : imageURLDesktop}
              />
            </Link>
          </div>
          {isMobile ? (
            <HamburgerMenu menu={menuOptions} />
          ) : (
            <NavigationMenu menu={menuOptions} />
          )}
        </Toolbar>
      </AppBar>
      {props.children}
    </React.Fragment>
  );
};

Navigation.propTypes = {};

export default Navigation;
