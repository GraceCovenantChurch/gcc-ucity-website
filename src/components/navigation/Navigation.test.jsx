import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import "../../__mocks__/matchMedia";
import { BrowserRouter as Router } from "react-router-dom";

import { render } from "@testing-library/react";
import Navigation from "./Navigation";

describe("Navigation", () => {
  test("matches the snapshot", () => {
    const { container: mobile } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 300 }}>
          <Navigation />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(mobile).toMatchSnapshot();

    const { container: desktop } = render(
      <Router>
        <ResponsiveContext.Provider value={{ width: 1000 }}>
          <Navigation />
        </ResponsiveContext.Provider>
      </Router>
    );
    expect(desktop).toMatchSnapshot();
  });
});
