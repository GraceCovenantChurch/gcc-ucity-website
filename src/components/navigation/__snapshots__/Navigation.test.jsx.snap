// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-appBarMobile-7 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-mobileToolBar-4 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-titleGrow-5"
      >
        <a
          href="/"
        >
          <img
            alt="nav-logo"
            class="makeStyles-menuMobileButton-3"
            src="test-file-stub"
          />
        </a>
      </div>
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-9 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;

exports[`Navigation matches the snapshot 2`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-appBarMobile-7 MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-mobileToolBar-4 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-titleGrow-5"
      >
        <a
          href="/"
        >
          <img
            alt="nav-logo"
            class="makeStyles-menuMobileButton-3"
            src="test-file-stub"
          />
        </a>
      </div>
      <button
        aria-label="menu"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-9 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </header>
</div>
`;
